lark/grammars/lark.lark

```diff
@@ -1,56 +1,70 @@
 # Lark grammar of Lark's syntax
 # Note: Lark is not bootstrapped, its parser is implemented in load_grammar.py
+# This grammar matches that one, but does not enfore some rules that it does.
+# If you want to enforce those, you can pass the "LarkValidator" over
+# the parse tree, like this:
+
+# from lark import Lark
+# from lark.lark_validator import LarkValidator
+#
+# lark_parser = Lark.open_from_package("lark", "grammars/lark.lark", parser="lalr")
+# parse_tree = lark_parser.parse(my_grammar)
+# LarkValidator.validate(parse_tree)
 
 start: (_item? _NL)* _item?
 
 _item: rule
      | token
      | statement
 
-rule: RULE rule_params priority? ":" expansions
-token: TOKEN token_params priority? ":" expansions
+rule: rule_modifiers RULE rule_params priority ":" expansions
+token: TOKEN priority? ":" expansions
```

Comment:
`priority` is already optional

Reply:
Yes, but `load_grammar.py` says [`priority` is a required element of `rule`](https://github.com/lark-parser/lark/blob/master/lark/load_grammar.py#L114), and that [`priority` is `_DOT NUMER` or null](https://github.com/lark-parser/lark/blob/master/lark/load_grammar.py#L117).  I wanted `lark.lark` to produce the same parse tree as `load_grammar.py`.

It's different for `token` (`term` in `load_grammar.py`) - there, `load_grammar.py` [says `priority` is optional[(https://github.com/lark-parser/lark/blob/master/lark/load_grammar.py#L162-L163).

Reply:
@erezsh If my comment of 2024-06-20 is acceptable, let's resolve this point.